lib/ansible/playbook/play.py

```diff
@@ -97,19 +98,39 @@ def __init__(self):
     def __repr__(self):
         return self.get_name()
 
+    def _validate_hosts(self, attribute, name, value):
+        # Only validate 'hosts' if a value was passed in to original data set.
+        if 'hosts' in self._ds:
+            err_msg = "Hosts list contains an invalid host value: '{host!s}'"
+
+            if not value:
+                raise AnsibleParserError("Hosts list cannot be empty. Please check your playbook")
+
+            if is_sequence(value):
+                # Make sure each item in the sequence is a valid string
+                for entry in value:
+                    if entry is None:
+                        raise AnsibleParserError("Hosts list cannot contain values of 'None'. Please check your playbook")
+                    elif not is_string(entry):
```

Comment:
I had to look up is_string. It's True for vaults, which aren't supported for hosts. The error for vaulted hosts is pretty clear at least though:

```
ERROR! the field 'hosts' should be a list of (<class 'str'>,), but the item 'host1' is a <class 'ansible.parsing.yaml.objects.AnsibleVaultEncryptedUnicode'>

The error appears to be in '/home/liveuser/ansible/test_hosts.yml': line 2, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
- name: test hosts
  ^ here
```

Reply:
Vaulted hosts. 🤯 I should add a test for that. Not something I thought about.

Reply:
I'll switch to an `issequence` check to error with `AnsibleVaultEncryptedUnicode` objects.